Errors of MSB3073 happen when executing command on folder that has space in it

<!-- If these statements apply, replace [ ] with [x] before filing your issue. -->
- [x] This issue is blocking <!-- Describe below what is blocked. -->
- [x] This issue is causing unreasonable pain

**NOTE:** This issue for me can be considered a bit "blocking" if we use arcade on Windows like those in .NET Winforms repo. It may not be applicable for repos that are not using `vswhere` or any other tool that depends on nuget's cache folder.
See description below.

<!-- Write your issue description below. -->
## Background

I was switching to new personal dev machine. Then I clone some `dotnet` 's repos to my machine.
When I build some repos such as winforms, I get this build error:

```
  System.Windows.Forms.UI.IntegrationTests -> C:\GITHUB-REPO\winforms\artifacts\bin\System.Windows.Forms.UI.IntegrationTests\Debug\net7.0\System.Windows.Forms.UI.IntegrationTests.dll
  'C:\Users\Eriawan' is not recognized as an internal or external command,
  operable program or batch file.
C:\Users\Eriawan Kusuma\.nuget\packages\microsoft.dotnet.cmake.sdk\7.0.0-beta.22206.10\build\Microsoft.DotNet.CMake.Sdk.targets(41,5): error MSB3073: The command "..." exited with code 9009. [C:\GITHUB-REPO\winforms\src\System.Windows.F
orms\tests\InteropTests\NativeTests\NativeTests.proj] 
```

## The cause of the problem

Looking at the binlog, I open the log section for NativeTests.proj then I found this:
![image](https://user-images.githubusercontent.com/8773147/162614271-42485957-1dbd-48e1-a032-218e6015a818.png)

Particularly, this line is the cause of the problem:
![image](https://user-images.githubusercontent.com/8773147/162614334-3b4b25a6-a5d0-495a-a56c-d42b06aa76b3.png)

## The proposed solution

On `Exec`, it executes that line of calling vswhere with the complete folder location that has space, and this will always result an error such as ` 'C:\Users\Eriawan' is not recognized as an internal or external command` because folder ` 'C:\Users\Eriawan'` will be treated as the folder location.
This happened because on this new machine, the default folder's name has a name in it, due to the default set by Windows. 

After I directly excape the Exec section from this:
```
    <Exec
      Command="$(VSWherePath) -latest -prerelease -products * -version $(VSGeneratorVersionRange) -requires $(_VSNativeCompilerComponentName) -property installationVersion"
      EchoOff="true"
      ConsoleToMsBuild="true"
      StandardOutputImportance="Low">
      <Output TaskParameter="ConsoleOutput" PropertyName="_HighestCompatibleVSVersion" />
    </Exec>
```

to this:
```
    <Exec
      Command="&quot;$(VSWherePath)&quot; -latest -prerelease -products * -version $(VSGeneratorVersionRange) -requires $(_VSNativeCompilerComponentName) -property installationVersion"
      EchoOff="true"
      ConsoleToMsBuild="true"
      StandardOutputImportance="Low">
      <Output TaskParameter="ConsoleOutput" PropertyName="_HighestCompatibleVSVersion" />
    </Exec>
```

Then the build works again. ~~I think I'm going to submit a PR on this issue, if the dotnet team agrees that this is a bug~~ 🙂 

EDIT:
I'm submitting PR now: #9015
Please review, and thanks!

--- a/src/Microsoft.DotNet.CMake.Sdk/build/Microsoft.DotNet.CMake.Sdk.targets
+++ b/src/Microsoft.DotNet.CMake.Sdk/build/Microsoft.DotNet.CMake.Sdk.targets
@@ -39,7 +39,7 @@
     <Error Condition="'$(_VSNativeCompilerComponentArchName)' == ''" Text="Unable to determine if Visual Studio has MSVC tools installed for the '$(Platform)' architecture." />
 
     <Exec
-      Command="$(VSWherePath) -latest -prerelease -products * -version $(VSGeneratorVersionRange) -requires $(_VSNativeCompilerComponentName) -property installationVersion"
+      Command="&quot;$(VSWherePath)&quot; -latest -prerelease -products * -version $(VSGeneratorVersionRange) -requires $(_VSNativeCompilerComponentName) -property installationVersion"
       EchoOff="true"
       ConsoleToMsBuild="true"
       StandardOutputImportance="Low">

